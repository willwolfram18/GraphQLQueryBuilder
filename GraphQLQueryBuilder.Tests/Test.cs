using GraphQLQueryBuilder.Tests.Models;
using Snapshooter.Xunit;
using Xunit;

namespace GraphQLQueryBuilder.Tests
{
    public class Test
    {
        [Fact]
        public void SimpleAddressQuery()
        {
            var query = new QueryBuilder<Address>()
                .AddQuery("address")
                .AddProperty(address => address.Street1)
                .AddProperty(address => address.Street2)
                .AddProperty(address => address.City)
                .AddProperty(address => address.State)
                .AddProperty(address => address.ZipCode)
                .Build();

            Snapshot.Match(query);
        }

        [Fact]
        public void RepeatedPropertiesIncludedInQuery()
        {
            var query = new QueryBuilder<Address>()
                .AddQuery("address")
                .AddProperty(address => address.Street1)
                .AddProperty(address => address.Street1)
                .AddProperty(address => address.Street1)
                .AddProperty(address => address.Street2)
                .AddProperty(address => address.City)
                .AddProperty(address => address.State)
                .AddProperty(address => address.ZipCode)
                .Build();

            Snapshot.Match(query);
        }
    }
}