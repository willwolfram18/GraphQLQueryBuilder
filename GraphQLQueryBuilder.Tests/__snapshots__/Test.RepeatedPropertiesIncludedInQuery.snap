query {
    address {
        Street1,
        Street1,
        Street1,
        Street2,
        City,
        State,
        ZipCode
    }
}